src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -252,13 +255,24 @@ private static string[] ProcessMetricUrls(string[] metricUrls, MetricsOptions me
                             ":",
                             metricUrl.Port.ToString(System.Globalization.CultureInfo.InvariantCulture),
                             metricUrl.PathBase);
+
+                        results.Warnings.Add($"Upgraded {metricUrl} to {metricUrls[i]} due to custom metrics. To host custom metrics over http " +
```

Comment:
This message should be moved to `LoggingExtensions` with a new `EventId`. If you take the structured information change in `AddressListenResults`, this becomes just adding a new instance of `UrlModification`.

Reply:
Optionally, the second part of the message ("To host custom metrics...") can be split into its own logging message so that if there is more than one upgraded URL we don't repeat the same message.